src/request.rs

```diff
@@ -45,14 +45,14 @@ pin_project_lite::pin_project! {
 
 impl Request {
     /// Create a new request.
-    pub fn new<U>(method: Method, url: U) -> Self
+    pub fn new<U>(method: Method, url: U) -> Result<Self, Error>
     where
         U: TryInto<Url>,
-        U::Error: std::fmt::Debug,
+        U::Error: std::fmt::Debug + Into<anyhow::Error>,
     {
-        let url = url.try_into().expect("Could not convert into a valid url");
+        let url = url.try_into().map_err(|e| Error::new(500, e))?;
```

Comment:
This does not need the `map_err` call to work:
```suggestion
        let url = url.try_into()?;
```
But if it's for clarity purposes, it's preferable to use the `Status` trait instead:
```suggestion
        let url = url.try_into().status(500)?;
```

Reply:
I got the following error after replacing map_err with status

```
no method named `status` found for enum `std::result::Result<url::Url, <U as std::convert::TryInto<url::Url>>::Error>` in the current scope

method not found in `std::result::Result<url::Url, <U as std::convert::TryInto<url::Url>>::Error>`

note: the method `status` exists but the following trait bounds were not satisfied:
      `<U as std::convert::TryInto<url::Url>>::Error: std::error::Error`
      which is required by `std::result::Result<url::Url, <U as std::convert::TryInto<url::Url>>::Error>: status::Status<url::Url, <U as std::convert::TryInto<url::Url>>::Error>`
      `<U as std::convert::TryInto<url::Url>>::Error: std::marker::Send`
      which is required by `std::result::Result<url::Url, <U as std::convert::TryInto<url::Url>>::Error>: status::Status<url::Url, <U as std::convert::TryInto<url::Url>>::Error>`
      `<U as std::convert::TryInto<url::Url>>::Error: std::marker::Sync`
      which is required by `std::result::Result<url::Url, <U as std::convert::TryInto<url::Url>>::Error>: status::Status<url::Url, <U as std::convert::TryInto<url::Url>>::Error>`
```